polynomial/core.py

```diff
@@ -143,6 +143,24 @@ def nth_derivative(self, n=1):
                 in zip(self, reversed(factors))]
         )
 
+    def integral(self, a, b):
+        """Return the integral of self from a to b."""
+        res = self.indefinite_integral
+        # Last term of indefinite integral is C.
+        res[0] = 0
+        return res.calculate(b) - res.calculate(a)
+
+    @property
+    def indefinite_integral(self):
+        """Return the polynomial object which is the integral of self."""
+        if not self:
+            return Polynomial("C")
+
+        return Polynomial(
+            [c/x for c, x in
+                zip(self, range(self.degree + 1, 0, -1))] + ["C"]
+            )
+
```

Comment:
I suggest adding a parameter for the value of the integration constant, since having the string "C" prevents the polynomial from being very useful for almost all operations and is maybe a not-so-desired result, as opposed to a plain *0* for example. Maybe having the parameter like `constant="C"` would cover that case, also there would be no need of lines 149 and 150 then.

Reply:
How would you cover the case of double integrals? Would we make constants a seperate polynomial that goes at the end?

Reply:
Well, currently we haven't implemented anything allowing operations between string and number coefficients, which would be needed for applying `indefinite_integral` to a polynomial ending with `"C"`. So three options come in my head in this scenario:
1. throwing an error (current)
or
2. implementing something like `nth_integral` which returns two things: the result without the constants and a polynomial with the constants
or
3. implementing a class `Integral` which handles this and can have a polynomial be easily extracted from it